product.go

```diff
@@ -83,6 +88,27 @@ type ParsedProductsResponse struct {
 	Data    []interface{} `json:"data"`
 }
 
+// ResourceTag represents a resource tag for a product in the Megaport Products API.
+type ResourceTag struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
```

Comment:
This would be better as a `map[string]string` I think, is there any reason besides API compat it was done this way?